test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/FormatterTest.scala

```diff
@@ -0,0 +1,258 @@
+    var width: Int = _
+    var precision: Int = _
+    var calls = 0
+
+    def formatTo(frm: Formatter, flags: Int, width: Int, precision: Int): Unit = {
+      this.calls += 1
+      this.flags = flags
+      this.width = width
+      this.precision = precision
+      frm.out().append("foobar")
+    }
+
+    def expectCalled(times: Int, flags: Int, width: Int, precision: Int): Unit = {
+      assertEquals(times, this.calls)
+      assertEquals(flags, this.flags)
+      assertEquals(width, this.width)
+      assertEquals(precision, this.precision)
+    }
+
+  }
+
+  def assertF(expected: String, format: String, args: AnyRef*): Unit = {
+    val fmt = new Formatter()
+    val res = fmt.format(format, args:_*).toString()
+    fmt.close()
+    assertEquals(expected, res)
+  }
+
+  def assertFC(expected: String, format: String, flags: Int, width: Int,
+      precision: Int): Unit = {
+    val fc = new FormattableClass
+    assertF(expected, format, fc)
+    fc.expectCalled(1, flags, width, precision)
+  }
+
+  def expectFormatterThrows[T <: Throwable](exeption: Class[T], format: String,
+      args: AnyRef*): Unit = {
+    val fmt = new Formatter()
+    assertThrows(exeption, fmt.format(format, args:_*))
+  }
+
+  // Explicitly define these as `var`'s to avoid any compile-time constant folding
+  var IntMax: Int = Int.MaxValue
+  var IntMin: Int = Int.MinValue
+  var ByteMax: Byte = Byte.MaxValue
+  var ByteMin: Byte = Byte.MinValue
+  var ShortMax: Short = Short.MaxValue
+  var ShortMin: Short = Short.MinValue
+
+  @Test def `should_provide_b_conversion`(): Unit = {
+    assertF("false", "%b", null)
+    assertF(JString.valueOf(true), "%b", true: JBoolean)
+    assertF(JString.valueOf(false), "%b", false: JBoolean)
+    assertF("true", "%b", new HelperClass)
+  }
+
+  @Test def `should_provide_h_conversion`(): Unit = {
+    val x = new HelperClass
+    assertF(Integer.toHexString(x.hashCode()), "%h", x)
+    assertF(Integer.toHexString(x.hashCode()).toUpperCase(), "%H", x)
+    assertF("null", "%h", null)
+  }
+
+  @Test def `should_provide_s_conversion`(): Unit = {
+    assertFC("foobar", "%s", 0, -1, -1)
+    if (!executingInJVM) {
+      // Issue #2246
+      // java.util.MissingFormatWidthException: %-s
+      assertFC("foobar", "%-s", FormattableFlags.LEFT_JUSTIFY, -1, -1)
+    }
+    assertFC("foobar", "%-10s", FormattableFlags.LEFT_JUSTIFY, 10, -1)
+    assertFC("foobar", "%#-10.2s",
+      FormattableFlags.LEFT_JUSTIFY | FormattableFlags.ALTERNATE, 10, 2)
+    assertFC("foobar", "%#10.2S",
+      FormattableFlags.UPPERCASE | FormattableFlags.ALTERNATE, 10, 2)
+    assertF("     hello", "%10s", "hello")
+    assertF("hello     ", "%-10s", "hello")
+    expectFormatterThrows(classOf[Exception], "%#s", "hello")
+  }
+
+  @Test def `should_provide_c_conversion`(): Unit = {
+    assertF("!    ", "%-5c", new Character('!'))
+  }
+
+  @Test def `should_provide_d_conversion`(): Unit = {
+    assertF("5", "%d",   new Integer(5))
```

Comment:
Double space after `,`. There are a bunch of those in this file, starting from this line. I guess they were vertical-aligned before and the swapping made this ugly. I suggest not to vertical-align anymore.